GitUI/BranchTreePanel/RepoObjectsTree.ContextActions.cs

```diff
@@ -116,12 +116,14 @@ private void ContextMenuBranchSpecific(ContextMenuStrip contextMenu)
                 return;
             }
 
-            var isNotActiveBranch = !node.IsActive;
-            _localBranchMenuItems.GetInactiveBranchItems().ForEach(t => t.Item.Visible = isNotActiveBranch);
+            _localBranchMenuItems.ForEach(t => t.Item.Visible = !areMultipleBranchesSelected // hide items if multiple branches are selected
+
+                // otherwise display all items for inactive (i.e. not current) branches and hide those applyin gonly to inactive branches for the active/current branch
```

Comment:
Have a look at it after I've applied some renaming and refactoring. It should be easier to understand now.

Reply:
I understand your preference is to disable unusable context menu items rather than hide them. Should I apply this pattern to all menu items then? Please note that those for local branches were hidden instead of disabled before.